### DISCUSSION TITLE
diskspace.plugin not showing / (root)

### DISCUSSION THREAD

**Discussion by fabriba** [netdata/netdata#13383] [Q&A] [answered]
Hi there, 
my web gui , under "DISKS" is showing many mount points, but not the one that is most important to me, the root.
Essentially I'd like to know how much disk space I am using.

I do see stats on reads/writes, but I am interested in how much space is used.

Thanks for any help you may provide :)

**Comment by thiagoftsm** [comment]
Hello @fabriba ,

How did you install your netdata? Did you modify `/etc/netdata/netdata.conf` to change plugin configuration? On my environment I did not change nothing, and this is what I have inside my configuration file:

```sh
[plugin:proc:diskspace]
        # remove charts of unmounted disks = yes
        # update every = 1
        # check for new mount points every = 15
        # exclude space metrics on paths = /proc/* /sys/* /var/run/user/* /run/user/* /snap/* /var/lib/docker/*
        # exclude space metrics on filesystems = *gvfs *gluster* *s3fs *ipfs *davfs2 *httpfs *sshfs *gdfs *moosefs fusectl autofs
        # space usage for all disks = auto
        # inodes usage for all disks = auto
```

As you can see, I m running default configuration, and I can see `/` on my host.

Best regards!

**Comment by fabriba** [comment]
Hi @[thiagoftsm] ,
I installed netdata on debian 11 with 
`sudo apt-get install netdata -y`

On my installation

`http://<IP:PORT>/netdata.conf 
`
shows a lot of informaion including the 
`[plugin:proc:diskspace]
`section you refer to, but I have almost nothing under  /etc/netdata/netdata.conf .

When I did add that particular section I did see some changes  in my webgui under  "DISK" in the right column, but under that I have `sda` , `/dev` etc etc, but I have no `/`.
The portion with `sda` seemed the most promising, but it doesn't show disk usage stats (in terms of "your disk is 70% full), only in terms of reads/writes per second.
The other partitions do show how full they are, but they are a bit meaningless to me as I did create only one partition on the installation disk.

I also tried to add
`[plugin:proc:diskspace:/]`
to my netdata.conf, but after service reload it was marked as unused under 
`http://<IP:PORT>/netdata.conf 
`
and still not showing up in the web gui.

**Reply by thiagoftsm** [reply]
Unless something has been changed, I think these packages installed directly with `apt-get` is not maintained by us, @Ferroin can you confirm this? 

Please, run the next command:

```sh
bash-5.1# netdata -version
netdata v1.35.0-191-nightly
```

I remember that years ago debian were delivering old versions of `netdata`, and due this users did not have fixes we merged.

**Reply by fabriba** [reply]
netdata v1.35.1

sorry for late reply, I was away for a couple of days.

**Reply by MrZammler** [reply]
Hi @fabriba 

Can you check your `error.log` if there are any errors from `DISKSPACE` ?

**Reply by fabriba** [reply]
found nothing recent, the logs dated 2022-07-12  are probably from to the time when 

> I also tried to add
> [plugin:proc:diskspace:/] 

(see first message)

```
root@debian:/home/fa# cat /var/log/netdata/error.log | grep DISKSPACE
root@debian:/home/fa# cat /var/log/netdata/error.log.1 | grep DISKSPACE
2022-07-12 00:33:53: netdata ERROR : PLUGIN[diskspace] : DISKSPACE: Cannot stat() mount point '/root/thinclient_drives' (disk '_root_thinclient_drives', filesystem 'fuse.xrdp-chansrv', root '/'). (errno 13, Permission denied)
2022-07-12 00:35:08: netdata ERROR : PLUGIN[diskspace] : DISKSPACE: Cannot stat() mount point '/home/fa/thinclient_drives' (disk '_home_fa_thinclient_drives', filesystem 'fuse.xrdp-chansrv', root '/'). (errno 13, Permission denied)
2022-07-15 01:44:56: netdata ERROR : PLUGIN[diskspace] : DISKSPACE: Cannot stat() mount point '/home/fa/thinclient_drives' (disk '_home_fa_thinclient_drives', filesystem 'fuse.xrdp-chansrv', root '/'). (errno 13, Permission denied)
root@debian:/home/fa#

```

**Reply by thiagoftsm** [reply]
Hello,

These errors that you are reporting brought some doubt, please, answer next question, and I hope after this we will have the final solution for you:

- What is your `netdata` permission?
```sh
bash-5.1# ls -l /usr/sbin/netdata
-rwxr-xr-x 1 root root 48201728 Jul 22 12:18 /usr/sbin/netdata
```
Diskspace is an internal plugin, so we need to verify `netdata` permission.

Best regards!

**Reply by fabriba** [reply]
am afraid it looks like yours
`
12:20:35 fa@debian ~ →  ls -l /usr/sbin/netdata
-rwxr-xr-x 1 root root 2868912 Jun 10 19:49 /usr/sbin/netdata
`

**Reply by fabriba** [reply]
to be clear, while I do some more reasearch... 
what I see is mostly consisted with [this](https://www.netdata.cloud/blog/how-to-monitor-your-disks-and-filesystems-now-also-with-ebpf/) page from netdata blogs ( I still don't have "/", but I do see disks stats about I/O like in that page).
What I would like to see is something like [this](https://www.how2shout.com/wp-content/uploads/2019/07/Charts-with-performance-information-for-all-the-system-disks-on-Windows.jpg) screenshot ([source](https://www.how2shout.com/how-to/how-to-install-netdata-on-windows-10-wsl.html)) , or [this](https://user-images.githubusercontent.com/16978632/64838335-00b5c000-d636-11e9-8861-60dcff49ef07.png) screenshot - except his problem of course ([source](https://github.com/netdata/netdata/issues/6839))

**Comment by fabriba** [comment]
@thiagoftsm (or anyone else) would you have any other suggestions? I am at loss here :/

**Reply by MrZammler** [reply]
Hi @fabriba ! We will try to debug this, we should have more information in a bit.!

**Reply by vlvkobal** [reply]
Please provide the contents of `/proc/self/mountinfo` and the output of `systemctl cat netdata`.

**Reply by vlvkobal** [reply]
Pay attention to https://github.com/netdata/netdata/issues/11498#issuecomment-950982878. I think the permissions for the netdata service might be the reason for the issue.

Could you set
```
[plugins]
        proc = no
        diskspace = yes
```
in `netdata.conf`, restart Netdata and check what mountpoints are available under the Disks menu item?

**Reply by fabriba** [reply]
1) sorry for late reply. I am on and off to another location and have had not much access to this box.

2) 
```
root@debian:/home/fa# cat /proc/self/mountinfo
21 28 0:19 / /sys rw,nosuid,nodev,noexec,relatime shared:7 - sysfs sysfs rw
22 28 0:20 / /proc rw,nosuid,nodev,noexec,relatime shared:13 - proc proc rw
23 28 0:5 / /dev rw,nosuid,relatime shared:2 - devtmpfs udev rw,size=3994640k,nr_inodes=998660,mode=755
24 23 0:21 / /dev/pts rw,nosuid,noexec,relatime shared:3 - devpts devpts rw,gid=5,mode=620,ptmxmode=000
25 28 0:22 / /run rw,nosuid,nodev,noexec,relatime shared:5 - tmpfs tmpfs rw,size=803884k,mode=755
28 1 8:2 / / rw,relatime shared:1 - ext4 /dev/sda2 rw,errors=remount-ro
29 21 0:6 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime shared:8 - securityfs securityfs rw
30 23 0:25 / /dev/shm rw,nosuid,nodev shared:4 - tmpfs tmpfs rw
31 25 0:26 / /run/lock rw,nosuid,nodev,noexec,relatime shared:6 - tmpfs tmpfs rw,size=5120k
32 21 0:27 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime shared:9 - cgroup2 cgroup2 rw,nsdelegate,memory_recursiveprot
33 21 0:28 / /sys/fs/pstore rw,nosuid,nodev,noexec,relatime shared:10 - pstore pstore rw
34 21 0:29 / /sys/firmware/efi/efivars rw,nosuid,nodev,noexec,relatime shared:11 - efivarfs efivarfs rw
35 21 0:30 / /sys/fs/bpf rw,nosuid,nodev,noexec,relatime shared:12 - bpf none rw,mode=700
36 22 0:31 / /proc/sys/fs/binfmt_misc rw,relatime shared:14 - autofs systemd-1 rw,fd=29,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=12796
37 21 0:7 / /sys/kernel/debug rw,nosuid,nodev,noexec,relatime shared:15 - debugfs debugfs rw
38 23 0:32 / /dev/hugepages rw,relatime shared:16 - hugetlbfs hugetlbfs rw,pagesize=2M
39 23 0:18 / /dev/mqueue rw,nosuid,nodev,noexec,relatime shared:17 - mqueue mqueue rw
40 21 0:11 / /sys/kernel/tracing rw,nosuid,nodev,noexec,relatime shared:18 - tracefs tracefs rw
41 21 0:33 / /sys/kernel/config rw,nosuid,nodev,noexec,relatime shared:19 - configfs configfs rw
42 21 0:34 / /sys/fs/fuse/connections rw,nosuid,nodev,noexec,relatime shared:20 - fusectl fusectl rw
87 28 8:1 / /boot/efi rw,relatime shared:44 - vfat /dev/sda1 rw,fmask=0077,dmask=0077,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro
685 25 0:43 / /run/user/116 rw,nosuid,nodev,relatime shared:399 - tmpfs tmpfs rw,size=803880k,nr_inodes=200970,mode=700,uid=116,gid=124
460 685 0:39 / /run/user/116/gvfs rw,nosuid,nodev,relatime shared:261 - fuse.gvfsd-fuse gvfsd-fuse rw,user_id=116,group_id=124
253 25 0:41 / /run/user/1000 rw,nosuid,nodev,relatime shared:194 - tmpfs tmpfs rw,size=803880k,nr_inodes=200970,mode=700,uid=1000,gid=1000
287 253 0:42 / /run/user/1000/gvfs rw,nosuid,nodev,relatime shared:248 - fuse.gvfsd-fuse gvfsd-fuse rw,user_id=1000,group_id=1000
root@debian:/home/fa#
```

3)
```
root@debian:/home/fa# systemctl cat netdata
# /lib/systemd/system/netdata.service
# netdata systemd target

[Unit]
Description=netdata - Real-time performance monitoring
Documentation=man:netdata
Documentation=file:///usr/share/doc/netdata/html/index.html
Documentation=https://github.com/netdata/netdata
After=network-online.target httpd.service squid.service nfs-server.service mysqld.service named.service postfix.service
ConditionPathExists=/etc/netdata/netdata.conf

[Service]
Type=simple
Environment="netdata_LOG_LOCATION=/var/log/netdata/log"
ExecStart=/usr/sbin/netdata -D
TimeoutStopSec=10
KillMode=mixed
KillSignal=SIGTERM
OOMScoreAdjust=-900

User=netdata
Group=netdata
Restart=on-abnormal
RestartSec=2s
LimitNOFILE=65536

WorkingDirectory=/tmp

# Hardening

NoNewPrivileges=false
PermissionsStartOnly=true
# CAP_SETGID is required for setgroups()
# CAP_NET_RAW is needed by fping, see #864370
CapabilityBoundingSet=CAP_DAC_READ_SEARCH CAP_SYS_PTRACE CAP_SETGID CAP_SETUID CAP_NET_RAW CAP_AUDIT_WRITE
PrivateTmp=true
ProtectHome=read-only
ProtectSystem=full

ReadOnlyDirectories=/
ReadWriteDirectories=/dev
ReadWriteDirectories=/proc/self
ReadWriteDirectories=/var/cache/netdata
ReadWriteDirectories=/var/lib/netdata
ReadWriteDirectories=/var/log
ReadWriteDirectories=/var/spool
ReadWriteDirectories=/run

RuntimeDirectory=netdata


# Access to devices and kernel modules and tunables is required
PrivateDevices=no
ProtectKernelModules=no
ProtectKernelTunables=no

[Install]
WantedBy=multi-user.target
```

**Reply by fabriba** [reply]
4) ` diskspace = yes` was already in `/etc/netdata/netdata.conf` under `[plugins]`.
I added `proc = no` and what happened is that from the right column, under `disks`, `sda` disappeared, while the rest of the mountpoints (such as `/dev`, `/tmp`, `/var`, some subdirectories, etc) remained seemingly unchanged.

**Reply by fabriba** [reply]
so re: 3. (above), it looks like I do match that [ #11498 (comment)](https://github.com/netdata/netdata/issues/11498#issuecomment-950982878.) you mentioned
```
ProtectHome=read-only
ProtectSystem=full
```

Not sure what to do from there tho, sorry... and thanks heaps for your help here!
@vlvkobal @MrZammler  (tagging you as I am aware took me 6 days to reply, so maybe you aren't following the thread after so many days)

**Reply by MrZammler** [reply]
Hi @fabriba 

Can you please try to edit `/lib/systemd/system/netdata.service`, comment `ReadOnlyDirectories=/` then do a `sudo systemctl daemon-reload` and then a `sudo systemctl restart netdata` ? And then check if you get the `/` under `Disks` ?

**Reply by fabriba** [reply]
I do, and I then removed `proc = no` that was making the overviews a little less interesting.
Thank you very much for your help!!!!!!!!!!! If you rewrite it as a new answer I will mark this answered.

As a last thing, which was what I was trying to achieve when I first started playing with this...
 is there any way to add `/` Disk space utilization as a gauge indicator to my overview at the top and/or at the top of my disks section?

**Reply by MrZammler** [reply]
Glad it worked! But it's far from a usable solution... We'll see how to make this better. Basically systemd marks `/` as Read only and we skip those when getting `Disk space utilization`. Not sure what the best solution is @vlvkobal , @ilyam8 ?

**Reply by vlvkobal** [reply]
We can add a configuration option with a list of mount points to show even if they are read-only. But that's far from the best solution as well.

**Comment by fabriba** [comment, marked as answer]
Workaround provided by MrZammier works:
```
 vi /lib/systemd/system/netdata.service #  comment ReadOnlyDirectories=/
sudo systemctl daemon-reload and 
sudo systemctl restart netdata
```

**Reply by fabriba** [reply]
(happy to give credit to MrZammier if you want to write an answer , I just thought for other users it'd be useful to see this is marked answered)

**Comment by ilyam8** [comment]
I see no `ReadOnlyDirectories` in our systemd unit files. 

> I installed netdata on debian 11 with
> sudo apt-get install netdata -y

What repository did you use?

**Reply by MrZammler** [reply]
> I see no `ReadOnlyDirectories` in our systemd unit files.
> 
> > I installed netdata on debian 11 with
> > sudo apt-get install netdata -y
> 
> What repository did you use?

Ilya, I believe it's this: https://packages.debian.org/sid/netdata

**Reply by ilyam8** [reply]
Oh, thanks! Then, I guess, we are ok with the workaround because our package has no that problem.

**Reply by fabriba** [reply]
FYI @MrZammler @ilyam8  I just purged the installation and reinstalled with
`wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh`

and am getting the same problem (I know how to work around it now, but this means it's not only occurring through debian packages)

**Reply by ilyam8** [reply]
> I just purged the installation

did you check that the systemd file was deleted?

> I know how to work around it now

And what is the workaround when `ReadOnlyDirectories` doesn't exist?
